Tests/Client/Unit Tests/Client+DevicesTests.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  Client+DevicesTests.swift
+//  StreamChatClientTests
+//
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import XCTest
+
+@testable import StreamChatClient
+
+class Client_DevicesTests: XCTestCase {
+
+    // Ideally, we would create a new client for every test case. Unfortunately, this is not technically
+    // possible at the moment.
+    lazy var client: Client = {
+        let sessionConfig = URLSessionConfiguration.default
+
+        sessionConfig.protocolClasses?.insert(RequestRecorderURLProtocol.self, at: 0)
+        Client.config = .init(apiKey: "test_api_key", defaultURLSessionConfiguration: sessionConfig)
+        return Client.shared
+    }()
+
+    func test_getDevice_createsRequest() {
```

Comment:
This is the first time I'm seeing test naming like this and I really liked it. Definitely more readable than `testGetDeviceCreatesRequest`
Is the format:
test -> This is a test (XCTest requirement)
getDevice -> endpoint that's being tested
createsRequest -> What is the expected result
?

Reply:
It's based on this tweet I saw **yesterday morning** right before starting working on this task https://twitter.com/qcoding/status/1255619642573426689?s=20 . 
First I was like ewww but then - it makes everything much more readable!

Reply:
Same for me! "eeewwwww....wait....wow!"
Thanks for sharing the tweet!

Reply:
I can't say I like it, but I'll agree with you if it useful for tests.